autofaiss/external/distributed.py

```diff
@@ -0,0 +1,372 @@
+"""
+Building the index with pyspark.
+"""
+import math
+import multiprocessing
+import os
+import shutil
+from functools import partial
+from multiprocessing.pool import ThreadPool
+from tempfile import NamedTemporaryFile
+from typing import List, Optional, Union, Iterator, Tuple, Callable, Any
+
+import faiss
+import fsspec
+import numpy as np
+import pandas as pd
+from fsspec import get_filesystem_class
+from fsspec.implementations.hdfs import PyArrowHDFS
+from tqdm import tqdm
+
+from autofaiss.readers.embeddings_iterators import get_file_list, get_matrix_reader
+from autofaiss.utils.cast import cast_memory_to_bytes
+from autofaiss.utils.decorators import Timeit
+
+
+def _yield_embeddings_batch(
+    embeddings_paths: List[str],
+    chunk_sizes: List[int],
+    file_system: fsspec.AbstractFileSystem,
+    start: int,
+    end: int,
+    embedding_column_name: str,
+    file_format: str,
+    id_columns: Optional[List[str]] = None,
+):
+    """Lazy load a batch of embeddings."""
+    if isinstance(file_system, PyArrowHDFS):
+        file_system = get_filesystem_class("hdfs")()
```

Comment:
Interesting! It didn't work without that?

Reply:
No, it didn't work within the executor. It's very werid.
But I use it outside the executor, everything is fine.

Reply:
ok, sounds like a path prefix and/or kerberos issue
it's ok to keep this code though